modules/ROOT/pages/batch-processing-concept.adoc

```diff
@@ -98,57 +87,81 @@ Though defined in context the above, it’s worth elaborating upon the terms _ba
+
+Note that it is possible to configure a Batch Step scope to filter the records to process or not within the scope. For details, see xref:batch-filters-and-batch-aggregator.adoc[] for more information.
+
+As a record is processed, it keeps track of each batch step it completes.
 
 image::batch-diagram.jpg[batch+diagram]
 
-Note that by default, a batch job instance does not wait for all its queued records to finish processing in one batch step before pushing any of them to the next batch step. Configuring an aggregator can change that behavior:
+[[batch_aggregator_processing]]
+By default, a batch job instance does not wait for all its queued records to finish processing in one batch step before pushing them to the next step. Configuring a Batch Aggregator scope (aggregator) changes this behavior:
 
-* Behavior with aggregator configured with fixed size
+* Behavior when the Batch Aggregator scope sets a fixed size:
 +
-In this scenario, the batch step sends the processed records to an aggregator, which starts processing the records and buffering them until the configured aggregator's size is reached. After that, the aggregator sends the aggregated records to the stepping queue.
+In this scenario, the aggregator pulls records from the stepping queue into arrays a fixed size. After processing records in an array, the aggregator sends the records back to the stepping queue for further downstream processing. A processor within the aggregator must be able to receive the array of records as input.
 +
 image:mruntime-batch-job-process-aggregator-fixed.png[Batch job process with Aggregator with fixed size]
 +
-<1> The batch job builds blocks of records limited by the configured block size and sends them to their corresponding batch step for processing. If fewer records are available than the configured size, the batch job sends them to the corresponding batch step instead of waiting for more records.
-<2> Each batch step receives one or more record blocks and starts processing them in parallel.
-<3> After the batch step processes a record, the batch step sends the record to the aggregator for further processing.
-<4> The aggregator continues processing records until the number of aggregated records reaches the configured aggregator's size.
-<5> The aggregator sends all the aggregated records to the stepping queue.
-
-* Behavior with aggregator configured for streaming
+<1> The Batch Job scope builds blocks of records limited by the block size configuration (100 per block, by default). This scope sends blocks of records from each batch job instance for processing within a batch step, the first batch step the scope contains if there is more than one step. If fewer records are available than the configured size, the Batch Job scope sends all records from a given batch job instance to the batch step instead of waiting for more to arrive.
+/////////
```

Comment:
/////////
//TODO: THIS MAY NOT BE 100% CLEAR.
// (1) CAN THE RECORDS IN THE BLOCK BE FROM A DIFFERENT INSTANCE? (I think not.)
// (2) WILL THE BATCH STEP EVER WAIT? FOR EXAMPLE, if the block size is 3 but there are 10 records, would _only the last block contain 1 record? Or is it possible for other blocks to contain fewer records than the set size. Everything depends on what's available in the stepping queue for a given instance when the step goes to pick up the records? (Sounds like the latter is true, but need to check.)
/////////